code/modules/overmap/contacts/_contacts.dm

```diff
@@ -53,10 +53,12 @@
 		images -= radar
 
 /datum/overmap_contact/proc/show()
-	for(var/weakref/W in owner?.viewers)
-		var/mob/M = W.resolve()
-		if(istype(M))
-			M.client?.images |= images
+	var/list/showing = (owner ? (owner.linked?.navigation_viewers || owner.viewers) : null)
+	if(length(showing))
+		for(var/weakref/W in showing)
+			var/mob/M = W.resolve()
+			if(istype(M))
+				M.client?.images |= images
```

Comment:
You might as well lift this client null check into the istype above, bit clearer that way imo.